helpers/clean-kibana-indexes.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Description: script to remove old .kibana* indexes' aliases created after Kibana
+# restart.
+#
+# The aim of the script is to keep only the latest index with the latest 
+# configuration, and delete the old ones.
+
+# Usage is: 
+# CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh"
+# CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" TENANT="drupalexample" ./clean-kibana-indexes.sh"
+
+set -eu
+
+# Set DEBUG variable to true, to start bash in debug mode
+DEBUG="${DEBUG:-"false"}"
+if [ "$DEBUG" = "true" ]; then
+        set -x
+fi
+
+# Cluster to connect to
+CLUSTER="${CLUSTER:-""}"
+if [ -z "$CLUSTER" ]; then
+	echo -e "Set CLUSTER variable\n"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh"
+	exit 1
+fi
+
+# Set the ACTION of the script between CLOSE or DELETE
+ACTION="${ACTION:-""}"
+if [ -z "$ACTION" ]; then
+	echo -e "Set ACTION variable to CLOSE or DELETE\n"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ACTION=CLOSE ./clean-kibana-indexes.sh"
+	exit 1
+fi
+
+# Set DRYRUN option to "true" to run in a dry-run mode
+DRYRUN="${DRYRUN:-"false"}"
+
+# Optional set a tenant to check and close/delete indexes
+TENANT="${TENANT:-"*"}"
+if [ "$TENANT" = "*" ]; then
+	REGEX="*"
+# This is a special tenant that requires a special REGEX
+elif [ "$TENANT" = "kibana" ]; then
+	REGEX=""
+else
+	REGEX="_*_$TENANT"
+fi
+
+# Switch context to the selected cluster
+echo -e "You're switching context to ${CLUSTER}"
+kubectl config use-context "${CLUSTER}"
+
+# Elasticsearch namespace of the cluster
+ELASTICSEARCH_NAMESPACE="${ELASTICSEARCH_NAMESPACE:-""}"
+
+if [ -z "$ELASTICSEARCH_NAMESPACE" ]; then
+	echo -e "\nSet ELASTICSEARCH_NAMESPACE variable"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh\n"
+	exit 1
+fi
+
+# Elasticsearch POD where run the script
+ELASTICSEARCH_POD="${ELASTICSEARCH_POD:-$(kubectl -n "$ELASTICSEARCH_NAMESPACE" get pods --no-headers -o custom-columns=NAME:.metadata.name|head -n1)}"
+echo -e "Selected POD to run commands is $ELASTICSEARCH_POD"
+
+# Generate script dynamically and pipe to the selected POD
+cat << EOF |
+#/bin/bash
+IFS=$'\n'
+
+# Generate array with kibana aliases for a specific tenant if specified, otherwise get all aliases
+ELASTICSEARCH_ALIASES=(\$(es-curl GET "_cat/aliases/.kibana$REGEX?h=alias,index"))
+
+# For each row, get current alias and index
+for ALIAS_ROW in "\${ELASTICSEARCH_ALIASES[@]}"
+do
+	ALIAS=\$(echo \$ALIAS_ROW|awk '{print \$1}')
+	INDEX=\$(echo \$ALIAS_ROW|awk '{print \$2}')
+	echo "\$ALIAS is the alias for \$INDEX"
+	
+	# Generate an arry with old indexes to remove
+	if [ "\$ALIAS" = ".kibana" ]; then
+		# This fix avoids the "catch-all" .kibana* regex
```

Comment:
you mention to avoid the `"catch-all" .kibana*` but the next line exactly uses that request, this is quite confusing, can you explain better what exactly happens here?

Reply:
So the full snippet is
```
# Generate an arry with old indexes to remove
 84         if [ "\$ALIAS" = ".kibana" ]; then
 85                 # This fix avoids the "catch-all" .kibana* regex
 86                 OLD_INDEXES=(\$(es-curl GET "_cat/indices/.kibana*?h=index" -s |grep -E '^.kibana_?[0-9]$'| grep -v -w \$INDEX))
 87         else
 88                 OLD_INDEXES=(\$(es-curl GET "_cat/indices/\$ALIAS*?h=index" -s |grep -v -w \$INDEX))
 89         fi
```
So for each row of the `ELASTICSEARCH_ALIASES` array, I get the ALIAS and the INDEX.
When we are going to close the `.kibana_*` indexes related to `.kibana` alias, I needed to find a way to avoid the but we had that `.kibana*` wildcard regex, matches **all** the indexes.
So I added this IF statement for the particular case of `.kibana` alias plus the `grep -E` part that does the trick.

Reply:
ok, but can you update the comment in the code to make this clearer how exactly this works? right now it looks like a bug.